packages/flutter/lib/src/material/scrollbar.dart

```diff
@@ -121,13 +122,18 @@ class _ScrollbarState extends State<Scrollbar> {
 
   void _onScrollStarted(ScrollableState scrollable) {
     _updateState(scrollable);
-   _fade.forward();
+    _scrollOffsetAnchor = _scrollOffset;
   }
 
   void _onScrollUpdated(ScrollableState scrollable) {
-    setState(() {
-      _updateState(scrollable);
-    });
+    _updateState(scrollable);
+    if (_scrollOffsetAnchor != _scrollOffset && _fade.value == 0.0 && !_fade.isAnimating) {
+      _fade.forward(); // Lazily start the scrollbar fade-in.
+    } else if (!_fade.isAnimating) {
+      setState(() {
+        // The scrollbar has faded in, rebuild it per the new scrollable state.
```

Comment:
you can call setState() with the _updateState call in both cases, and avoid the second branch here, because it's harmless to call setState if fade _is_ animating.